lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -659,7 +659,11 @@ protected function prepareUpdateData($entity, bool $isInsert = false)
                     continue;
                 }
 
-                $this->columnTypes[$columnName] = $fieldMapping['type'];
+                if ($newVal === null && $fieldMapping['type'] === Types::BLOB) {
+                    $this->columnTypes[$columnName] = Types::STRING;
```

Comment:
This feels like a hack, tbh. I did not really dig into the issue yet. I'd feel better if we bound the value with `ParameterType::NULL` instead. Would that be possible?

Reply:
I believe null values should be bound as `ParameterType::NULL` regardless of their type.